src/notebook.c

```diff
@@ -65,6 +66,8 @@ static gboolean switch_in_progress = FALSE;
 static GtkWidget *switch_dialog = NULL;
 static GtkWidget *switch_dialog_label = NULL;
 
+static gboolean doc_saves_to_new_file = FALSE;
```

Comment:
meh, globals are evil :(
even a static variable passed through as user data to the callbacks would be less ugly :]